app/scripts/migrations/067.js

```diff
@@ -0,0 +1,68 @@
+import { cloneDeep } from 'lodash';
+import BigNumber from 'bignumber.js';
+import { TEST_CHAINS } from '../../../shared/constants/network';
+
+const hexNumberIsGreaterThanZero = (hexNumber) =>
+  new BigNumber(hexNumber || '0x0', 16).gt(0);
+
+const version = 67;
+
+/**
+ * Sets the showTestNetworks property to true if it was false or undefined, and there is evidence
+ * that the user has used a test net
+ */
+export default {
+  version,
+  async migrate(originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData);
+    versionedData.meta.version = version;
+    const state = versionedData.data;
+    const newState = transformState(state);
+    versionedData.data = newState;
+    return versionedData;
+  },
+};
+
+function transformState(state) {
+  let newState = state;
+
+  const PreferencesController = state?.PreferencesController || {};
+  const preferences = PreferencesController.preferences || {};
+
+  if (!preferences.showTestNetworks) {
```

Comment:
Instead of this huge conditional, can we do `if(preferences.showTestNetworks) { return state }`?

Reply:
done in 6c9becbfa